src/GitVersion.Core.Tests/IntegrationTests/RemoteRepositoryScenarios.cs

```diff
@@ -98,6 +99,24 @@ public void GivenARemoteGitRepositoryWhenCheckingOutDetachedHeadUsingExistingImp
             $"It looks like the branch being examined is a detached Head pointing to commit '{fixture.LocalRepositoryFixture.Repository.Head.Tip.Id.ToString(7)}'. Without a proper branch name GitVersion cannot determine the build version.");
     }
 
+    [Test]
+    [TestCase(true)]
+    [TestCase(false)]
+    public void GivenARemoteGitRepositoryWhenCheckingOutDetachedHeadUsingExistingImplementationHandleDetachedBranch(bool handleDetachedBranch)
+    {
+        using var fixture = new RemoteRepositoryFixture();
+        Commands.Checkout(
+            fixture.LocalRepositoryFixture.Repository,
+            fixture.LocalRepositoryFixture.Repository.Head.Tip);
```

Comment:
Please avoid the use of `Commands.Checkout()`.

```suggestion
        fixture.LocalRepositoryFixture.Checkout(fixture.LocalRepositoryFixture.Repository.Head.Tip);
```